cadc-rest/src/main/java/ca/nrc/cadc/rest/SyncInput.java

```diff
@@ -239,107 +237,97 @@ public String getParameter(String name)
         if (request.getMethod().equals("GET") ||
-                request.getMethod().equals("HEAD") ||
-                request.getMethod().equals("DELETE"))
-        {
+            request.getMethod().equals("HEAD") ||
+            request.getMethod().equals("DELETE")) {
             Enumeration<String> names = request.getParameterNames();
-            while (names.hasMoreElements())
-            {
+            while (names.hasMoreElements()) {
                 String name = names.nextElement();
-                processParameter(name, request.getParameterValues(name));
+                processParameter(name, Arrays.asList(request.getParameterValues(name)));
             }
-        }
-        else
-        {
+        } else {
             String contentType = request.getContentType();
-            if (contentType != null)
-            {
+            if (contentType != null) {
                 int i = contentType.indexOf(';');
-                if (i > 0)
+                if (i > 0) {
                     contentType = contentType.substring(0, i);
+                }
             }
             log.debug("Content-Type: " + contentType);
-            if (contentType != null && contentType.equalsIgnoreCase(RestAction.URLENCODED))
-            {
+            if (contentType != null && contentType.equalsIgnoreCase(RestAction.URLENCODED)) {
                 Enumeration<String> names = request.getParameterNames();
-                while (names.hasMoreElements())
-                {
+                while (names.hasMoreElements()) {
                     String name = names.nextElement();
-                    processParameter(name, request.getParameterValues(name));
+                    processParameter(name, Arrays.asList(request.getParameterValues(name)));
                 }
-            }
-            else if (contentType != null && contentType.startsWith(RestAction.MULTIPART))
-            {
-            	try
-            	{
+            } else if (contentType != null && contentType.startsWith(RestAction.MULTIPART)) {
+                try {
                     ServletFileUpload upload = new ServletFileUpload();
                     FileItemIterator itemIterator = upload.getItemIterator(request);
                     processMultiPart(itemIterator);
-            	}
-            	catch(FileUploadException ex)
-            	{
+                } catch (FileUploadException ex) {
                     throw new IOException("Failed to process " + RestAction.MULTIPART, ex);
-            	}
-            }
-            else
-            {
+                }
+            } else {
                 processStream(null, contentType, request.getInputStream());
             }
         }
     }
 
-    private void processParameter(String name, String[] values)
-    {
-    	this.params.put(name, Arrays.asList(values));
+    private void processParameter(String name, List<String> values) {
+        this.params.put(name, values);
```

Comment:
I would put the code to get the existing List from the params map here, eg:
List<String> cur = params.get(name);
```if (cur == null) 
  params.put(name, values);
else 
 cur.addAll(values);```

and then simplify the processMultipart. It is more efficient and changes to init() or processMultipart are less likely to re-introduce the bug.